### DISCUSSION TITLE
flopy GBOB package

### DISCUSSION THREAD

**Discussion by zargol7419** [modflowpy/flopy#1499] [Q&A] [answered]
Hi, I am loading an existing modflow model using this code:  flopy.modflow.Modflow.load()
and I encounter this error while loading GBOB package. How can I solve this problem?

![image](https://user-images.githubusercontent.com/103249900/184479104-7749721b-5294-497e-8c6c-42183c577f16.png)

**Comment by wpbonelli** [comment]
Hi @zargol7419, it looks like this may be a [typo](https://github.com/modflowpy/flopy/blob/develop/flopy/modflow/mfflwob.py#L605) in the head-dependent flow-boundary observation package (`GOBO` should be `GBOB`). Thanks for bringing it to our attention.

Would you mind sharing more of the code that produces this error? That would help to reproduce and fix it.

Also, it looks like you have a conda environment with Python 2.7. If you can, it is probably best to upgrade to Python >= 3.7 and the current version of FloPy (3.3.5).

**Comment by zargol7419** [comment, marked as answer]
Hi dear @w-bonelli, thanks for your answer,
Actually it is the first line of my code and the whole of it cause I couldn't write it totally after getting this error.
My floppy version is 3.3.5 (the latest one) and I have tried python 3.7 but there was no difference in the result but I am going to upgrade my python. So how can I fix this problem independently?

**Reply by wpbonelli** [reply]
FloPy 3.3.5 is not compatible with Python 2, so you will need upgrade to Python 3.7 or greater.

If I understand correctly, you have invoked `flopy.modflow.Modflow.load()` with no arguments. You must specify a namefile when using `Modflow.load`. See the [example scripts and notebooks](https://github.com/modflowpy/flopy/blob/master/examples/Notebooks/flopy3_working_stack_demo.ipynb) for a demonstration of how to load and run a model. If you would like to load an existing dataset, you can create a modflow object with `flopy.modflow.Modflow()`, then use the `load` method on the appropriate package, passing the modflow object and the path to the data file as arguments. See [this notebook](https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_SFR2_load.ipynb) for an example.

I will follow up regarding the potential typo in `ModflowFlwob`.

**Reply by zargol7419** [reply]
exactly, I wanna load an existing model which I have made before and then wanna change the content of wel package based on my optimization code and finally run the modflow model again.  It is possible in flopy package right? 
I am a kind of beginer in using python that's why my problem may seem simple.
I have changed the content of source code in my library fileand it worked. my model has been loaded. I really appreciate your answer.
Thank you so much for your cooperation

**Reply by wpbonelli** [reply]
No problem, glad you solved the issue.

Yes, it is possible to use your own modified or custom packages with FloPy. The package/plugin mechanism is still under development and minimally documented, but it can be used now. To see how it works, take a look at how FloPy uses it internally for the wel package and other built-ins. Look out for more in forthcoming releases later this year or early next year.
